libr/core/core_annotated_code.c

```diff
@@ -0,0 +1,255 @@
+			offset_width = 4;
+		}
+	}
+
+	RCons *cons = r_cons_singleton();
+	RCodeAnnotation *annotation;
+	r_vector_foreach (&code->annotations, annotation) {
+		if (annotation->type != R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT) {
+			continue;
+		}
+
+		// (1/3)
+		// now we have a syntax highlighting annotation.
+		// pick a suitable color for it.
+		const char* color = Color_RESET;
+		switch (annotation->syntax_highlight.type) {
+		case R_SYNTAX_HIGHLIGHT_TYPE_COMMENT:
+			color = PALETTE(comment): Color_WHITE;
+			break;
+		case R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD:
+			color = PALETTE(pop): Color_MAGENTA;
+			break;
+		case R_SYNTAX_HIGHLIGHT_TYPE_DATATYPE:
+			color = PALETTE(func_var_type): Color_BLUE;
+			break;
+		case R_SYNTAX_HIGHLIGHT_TYPE_FUNCTION_NAME:
+			color = PALETTE(fname): Color_RED;
+			break;
+		case R_SYNTAX_HIGHLIGHT_TYPE_CONSTANT_VARIABLE:
+			color = PALETTE(num): Color_YELLOW;
+		default:
+			break;
+		}
+
+		// (2/3)
+		// the chunk before the syntax highlighting annotation should not be colored
+		for (; cur < annotation->start && cur < len; cur++) {
+			// if we are starting a new line and we are printing with offsets
+			// we need to prepare the bar with offsets on the left handside before that
+			if (line_offsets && (cur == 0 || code->code[cur - 1] == '\n')) {
+				ut64 offset = 0;
+				if (line_idx < line_offsets->len) {
+					offset = *(ut64 *)r_vector_index_ptr(line_offsets, line_idx);
+				}
+				print_offset_in_binary_line_bar(code, offset, offset_width);
+				line_idx++;
+			}
+			r_cons_printf("%c", code->code[cur]);
+		}
+
+		// (3/3)
+		// everything in between the "start" and the "end" inclusive should be highlighted
+		PRINT_COLOR (color);
+		for (; cur < annotation->end && cur < len; cur++) {
+			// if we are starting a new line and we are printing with offsets
+			// we need to prepare the bar with offsets on the left handside before that
+			if (line_offsets && (cur == 0 || code->code[cur - 1] == '\n')) {
+				ut64 offset = 0;
+				if (line_idx < line_offsets->len) {
+					offset = *(ut64 *)r_vector_index_ptr(line_offsets, line_idx);
+				}
+				PRINT_COLOR (Color_RESET);
+				print_offset_in_binary_line_bar(code, offset, offset_width);
+				PRINT_COLOR (color);
+				line_idx++;
+			}
+			r_cons_printf("%c", code->code[cur]);
+		}
+		PRINT_COLOR (Color_RESET);
+	}
+	// the rest of the decompiled code should be printed
+	// without any highlighting since we don't have any annotations left
+	for (; cur < len; cur++) {
+		// if we are starting a new line and we are printing with offsets
+		// we need to prepare the bar with offsets on the left handside before that
+		if (line_offsets && (cur == 0 || code->code[cur - 1] == '\n')) {
+			ut64 offset = 0;
+			if (line_idx < line_offsets->len) {
+				offset = *(ut64 *)r_vector_index_ptr(line_offsets, line_idx);
+			}
+			print_offset_in_binary_line_bar(code, offset, offset_width);
+			line_idx++;
+		}
+		r_cons_printf("%c", code->code[cur]);
+	}
+}
+
```

Comment:
no need for all these spaces

Reply:
I have formatted the code in the new commit. Now when I use`./sys/clang-format-diff.py`, it doesn't show any formatting problems. Please check if there are any problems with indentation now.